### PR TITLE
chore: fix circular dependencies related to assets.ts

### LINKED ISSUE #1340: chore: fix circular dependencies between assets modules
### Describe what should be investigated or refactored
The assets modules have several circular dependencies between them. These should be fixed.

### Additional context
Add any other context or screenshots about the technical debt here.

### PR DESCRIPTION
**Pull request by schaeferka**
## Description
The assets modules have several circular dependencies between them. These should be fixed.
...

End to End Test:    

## Related Issue

Fixes #1340 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [ ] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [ ] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/lib/assets/assetsDeployer.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/* eslint-disable complexity */
```
Comment (schaeferka) [inline src/lib/assets/assetsDeployer.ts]:
```suggestion
```

File: src/lib/assets/assetsDeployer.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+/* eslint-disable max-statements */
```
Comment (schaeferka) [inline src/lib/assets/assetsDeployer.ts]:
```suggestion
```

File: src/lib/assets/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { createAssets } from "./index";
+import { beforeEach, describe, expect, it, jest } from "@jest/globals";
+
+// Mock dependencies
+jest.mock("./assetsConfig");
+jest.mock("./assetsDeployer");
+
+describe("createAssets", () => {
+  const mockConfig = {
+    // Populate with necessary mock properties
```
Comment (schaeferka) [inline src/lib/assets/index.test.ts]:
```suggestion
```

File: src/lib/finalizer.ts [hunk 4]
```diff
@@ -26,14 +26,17 @@ export function addFinalizer<K extends KubernetesObject>(request: PeprMutateRequ
   }
 
   request.Merge({ metadata: { finalizers } } as DeepPartial<K>);
+
+  // Log the addition of a finalizer with the alias
+  Log.info({ alias: request.Request.name }, `Added finalizer '${peprFinal}' to ${request.Raw.metadata?.name}`);
```
Comment (cmwylie19) [inline src/lib/finalizer.ts]: Does this provide value to the user? We are trying to cut our logs down
https://github.com/defenseunicorns/pepr/issues/1360

File: src/cli/build.ts [hunk 5]
```diff
@@ -180,9 +176,9 @@ export default function (program: RootCmd) {
 
       let zarf = "";
       if (opts.zarf === "chart") {
-        zarf = assets.zarfYamlChart(chartPath);
+        zarf = await assets.zarfYamlChart(chartPath);
       } else {
-        zarf = assets.zarfYaml(yamlFile);
+        zarf = await assets.zarfYaml(yamlFile);
```
Comment (cmwylie19) [inline src/cli/build.ts]:
```suggestion
        zarf = await assets.zarfYaml(yamlFile);
```

Why is this async?

File: src/cli/build.ts [hunk 6]
```diff
@@ -180,9 +176,9 @@ export default function (program: RootCmd) {
 
       let zarf = "";
       if (opts.zarf === "chart") {
-        zarf = assets.zarfYamlChart(chartPath);
+        zarf = await assets.zarfYamlChart(chartPath);
```
Comment (cmwylie19) [inline src/cli/build.ts]: this should not be async

**Review by cmwylie19 — CHANGES_REQUESTED** [review]
Still need to clean up some async functions, needs further review. looks good though

File: src/cli/build.ts [hunk 7]
```diff
@@ -111,7 +103,9 @@ export default function (program: RootCmd) {
         // only actually build/push if there are files to include
         if (includedFiles.length > 0) {
           await createDockerfile(cfg.pepr.peprVersion, cfg.description, includedFiles);
-          execSync(`docker build --tag ${image} -f Dockerfile.controller .`, { stdio: "inherit" });
+          execSync(`docker build --tag ${image} -f Dockerfile.controller .`, {
```
Comment (cmwylie19) [inline src/cli/build.ts]: What happened to the customImage flag?

File: src/lib/assets/assetsDeployer.ts [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+    // Namespace validation and generating YAMLs
+    for (const capability of this.assetsConfig.capabilities) {
+      namespaceComplianceValidator(capability, this.assetsConfig.alwaysIgnore?.namespaces);
+    }
+
+    return allYaml(this.assetsConfig, imagePullSecret);
+  }
+
+  async zarfYaml(path: string) {
```
Comment (cmwylie19) [inline src/lib/assets/assetsDeployer.ts]: doesn't this just return some json?

File: src/lib/assets/assetsDeployer.ts [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    return zarfYaml(this.assetsConfig, path);
```
Comment (cmwylie19) [inline src/lib/assets/assetsDeployer.ts:47]: doesn't this just return some json?

File: src/lib/assets/assetsDeployer.ts [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  async zarfYamlChart(path: string) {
```
Comment (cmwylie19) [inline src/lib/assets/assetsDeployer.ts]: doesn't this just return some json? not async needed unless i am missing something

**Comment by cmwylie19** [comment]
this also breaks customImage and image in the build cli command

File: src/lib/finalizer.ts (hunk 4 above)
Comment (schaeferka) [reply src/lib/finalizer.ts]: Removed the log message

File: src/cli/build.ts (hunk 5 above)
Comment (schaeferka) [reply src/cli/build.ts]: The return value is a Promise<string> so the await is needed or you get:

`Type 'Promise<string>' is not assignable to type 'string'.ts(2322)`

File: src/cli/build.ts (hunk 6 above)
Comment (schaeferka) [reply src/cli/build.ts]: Fixed these by making the functions in assetsDeployer.ts no be async.

File: src/cli/build.ts (hunk 5 above)
Comment (schaeferka) [reply src/cli/build.ts]: Fixed these by making the functions in assetsDeployer.ts no be async.

File: src/lib/assets/assetsDeployer.ts (hunk 8 above)
Comment (schaeferka) [reply src/lib/assets/assetsDeployer.ts]: Fixed.

File: src/lib/assets/assetsDeployer.ts (hunk 10 above)
Comment (schaeferka) [reply src/lib/assets/assetsDeployer.ts]: Fixed.

File: src/cli/build.ts (hunk 7 above)
Comment (schaeferka) [reply src/cli/build.ts]: I removed it by accident when I was fixes merge conflicts yesterday.

**Comment by samayer12** [comment]
Closing. We're gonna revisit the concept of #1340 after `0.39.0` now that we've identified some issues within coverage from our test suite (see #1396).

**Comment by samayer12** [comment]
Closing. We're gonna revisit the concept of #1340 after `0.39.0` now that we've identified some issues within coverage from our test suite (see #1396).

**Pull request closed without merging**